tls_test.go

```diff
@@ -93,6 +93,11 @@ func TestTLSHandshake(t *testing.T) {
 			t.Fatalf("expected to handshake a TLS connection, got err: %v", err)
 		}
 	}
+
+	st := c.ConnectionState()
+	if !st.HandshakeComplete {
+		t.Errorf("expected ConnectionState, got: %+v", st)
```

Comment:
This error message doesn't actually warn the user that there was a TLS handshake failure. Please change this to "TLS handshake failed, TLS connection state: …".